pkg/operator/controller.go

```diff
@@ -154,6 +156,29 @@ func (c *Controller) sync() error {
 		return err
 	}
 
+	// 4.2 - on upgrade, the trusted CA bundle may not be present, and the networking operator may
+	// take a long time to inject the CA bundle.
+	// In this situation the operator should report itself as at level.
+	// Later, when the trusted CA bundle is present, we proceed as usual and upgrade the operands.
+	if !c.hasTrustedCA() {
+		updateCondition(cr, operatorapi.OperatorStatusTypeAvailable, operatorapi.OperatorCondition{
+			Status:  operatorapi.ConditionTrue,
+			Reason:  "TrustedCANotPresent",
+			Message: "Inactive due to trusted CA bundle not present.",
+		})
+		updateCondition(cr, operatorapi.OperatorStatusTypeProgressing, operatorapi.OperatorCondition{
+			Status:  operatorapi.ConditionFalse,
+			Reason:  "TrustedCANotPresent",
+			Message: "Inactive due to trusted CA bundle not present.",
+		})
+		updateCondition(cr, operatorapi.OperatorStatusTypeDegraded, operatorapi.OperatorCondition{
+			Status:  operatorapi.ConditionFalse,
```

Comment:
True?

Reply:
I need to refine this a bit more - we in fact should report `Degraded=true` if we don't have CA's after a specific period of time (an hour perhaps).